modules/date-picker/main/index.coffee

```diff
@@ -300,29 +331,32 @@ buildDatepicker = (datepicker) ->
 
 
 # Shared Functions for both picker types
-setupInput = (datepicker) ->
+setupInput = (datepicker, initial) ->
   _ = datepicker._
 
   if datepicker.options.selectRange
     range = datepicker.range()
     _.inputStart.value(datepicker.localizer.date range.start, _.useInbuilt)
     _.inputEnd.value(datepicker.localizer.date range.end or range.start, _.useInbuilt)
   else
-    _.input.value(datepicker.localizer.date datepicker.date(), _.useInbuilt)
+    if not (datepicker.options.v2Features and initial)
+      _.input.value(datepicker.localizer.date(datepicker.date(), _.useInbuilt))
 
 
 # Function for updating the input fields and emitting the change event.
-updateDatepicker = (datepicker, suppress) ->
+updateDatepicker = (datepicker, suppress, initial) ->
   _ = datepicker._
   validateDates(datepicker)
   if not _.preventFeedback
     _.preventFeedback = true
     if datepicker.options.selectRange
       _.inputStart.classed('hx-date-error', false)
       _.inputEnd.classed('hx-date-error', false)
+    else if _.inputOnlyMode
+      datepicker.options.dateValidationChange?(true)
     else
       _.input.classed('hx-date-error', false)
-    setupInput datepicker
+    setupInput datepicker, initial
```

Comment:
When the datepicker is created, it should not set the `input` value so we use this `initial` flag to prevent it being set